src/wp-includes/interactivity-api/class-wp-interactivity-api.php

```diff
@@ -295,6 +297,11 @@ private function process_directives_args( string $html, array &$context_stack, a
 			 * We still process the rest of the HTML.
 			 */
 			if ( 'SVG' === $tag_name || 'MATH' === $tag_name ) {
+				if ( $p->get_attribute_names_with_prefix( 'data-wp-' ) ) {
```

Comment:
I suppose it's possible for a directive to be used on a tag _inside_ of an `svg` or `math` element tree, in which case the directives are ignored _and_ there is no such notice displayed, correct? 

I suppose the only way to deal with that would be to add this same condition inside of `skip_to_tag_closer`, for example:

```diff
diff --git a/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php b/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php
index 3b2dcb12..ef04ac92 100644
--- a/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php
+++ b/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php
@@ -204,6 +204,11 @@ final class WP_Interactivity_API_Directives_Processor extends WP_HTML_Tag_Proces
 				'tag_closers' => 'visit',
 			)
 		) ) {
+			if ( ! $this->is_tag_closer() && $this->get_attribute_names_with_prefix( 'data-wp-' ) ) {
+				/* translators: 1: SVG or MATH HTML tag, 2: Namespace of the interactive block. */
+				$message = sprintf( __( 'Interactivity directives were detected on an incompatible %1$s tag when processing "%2$s". These directives will be ignored.' ), $tag_name, end( $namespace_stack ) );
+				_doing_it_wrong( __METHOD__, $message, '6.6.0' );
+			}
 			if ( $this->has_self_closing_flag() ) {
 				continue;
 			}
```

Reply:
I think I tried to check the content in https://github.com/WordPress/wordpress-develop/pull/6413/commits/9a172ec2685bd61db62f291e0adcd25ea7b29b7d but was getting the entire root block and I reverted in https://github.com/WordPress/wordpress-develop/pull/6413/commits/eff4f6ccce23548168267f8310bdce889ff6397b

As far as I remember, inner tags were not processed. But, as I reread, it seems that if there are no directives in the main SVG or MATH tags, but there are tags inside those elements, they won't be processed and there won't be any notice.

I'll add a test for that and update the code with your proposal. Thanks!!

Reply:
It seems that is not so easy to determine the content inside an SVG tag. So checking if it has any directive would not be possible.

HTML API will support SVG in a future, so we may not need this notice.

Discussed in the public Slack channel: 

https://wordpress.slack.com/archives/C071CRKGKUP/p1716839075325949

Reply:
I'm not sure I understand, as it is currently walking over all the content inside the SVG tag in the `skip_to_tag_closer` method. But if the intention is that SVG will be supported soon anyway, then I guess it doesn't make sense to add a warning.

Reply:
I could not reach the inner content inside that function, is there something I may be missing?

ping @sirreal 

![Screenshot 2024-05-29 at 16 02 54](https://github.com/WordPress/wordpress-develop/assets/37012961/997cf90d-5a57-493f-8b4c-9506bab7598f)

Reply:
It does look like you should be able to reach all the tags there and could show debug messages. The issue in the screenshot seems to be with the `next_tag()` query, which is searching for the next tag with the matching name (`SVG`). Something like this should visit all the tags and provide a chance to show the message there:

```diff
diff --git before/src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php after/src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php
index b12dcb4b3b1..2e15f583136 100644
--- before/src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php
+++ after/src/wp-includes/interactivity-api/class-wp-interactivity-api-directives-processor.php
@@ -199,15 +199,17 @@ final class WP_Interactivity_API_Directives_Processor extends WP_HTML_Tag_Proces
 		$depth    = 1;
 		$tag_name = $this->get_tag();
 		while ( $depth > 0 && $this->next_tag(
-			array(
-				'tag_name'    => $tag_name,
-				'tag_closers' => 'visit',
-			)
+			array( 'tag_closers' => 'visit' )
 		) ) {
-			if ( $this->has_self_closing_flag() ) {
-				continue;
+			if ( ! $this->is_tag_closer() && ! empty( $this->get_attribute_names_with_prefix( 'data-wp-' ) ) ) {
+				// show warning
+			}
+			if ( $this->get_tag() === $tag_name ) {
+				if ( $this->has_self_closing_flag() ) {
+					continue;
+				}
+				$depth += $this->is_tag_closer() ? -1 : 1;
 			}
-			$depth += $this->is_tag_closer() ? -1 : 1;
 		}
 
 		return 0 === $depth;
```

Reply:
That code worked like a charm. Thanks for providing it.
Now we differentiate between SVG and MATH tags and their inner tags with slightly different messages.
All tests has been updated.